assembly/assembly-ide-war/src/main/webapp/_app/activity.js

```diff
@@ -58,7 +65,28 @@ var ActivityTracker = new function () {
             }
         };
         request.open("PUT", ActivityTracker.url, true);
+        if (csrfToken) {
+            request.setRequestHeader("X-CSRF-Token", csrfToken);
+        }
         request.send();
     };
 
+    this.requestCsrfToken = function(restContext, tokenPresentHandler, tokenMissingHandler) {
```

Comment:
Why not obtain token in extension using RequestFactory and just pass as init param?